### PR TITLE
Persists selected music library folder when changed through the GUI

### LINKED ISSUE #453: Elementary Music forgets selected music library folder when changed through GUI
Hi,

I noticed, strange behavior in Elementary Music player.
If I change music library location from GUI, then exit the app. On next launch, elementary has set previous location, not the one I picked through GUI.

If I change /io/elementary/music/settings/music-folder using dconf-editor, the it works on next launch.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/65690702-elementary-music-forgets-selected-music-library-folder-when-changed-through-gui?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60766398&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ZavierHenry**
Resolves #453 and #295 by saving the music folder location that was changed through the API

### CODE REVIEW

File: src/LocalBackend/LocalLibrary.vala [hunk 1]
```diff
@@ -198,6 +198,8 @@ public class Noise.LocalLibrary : Library {
 
             set_music_folder_thread.begin (folder);
         }
+        
+        Settings.Main.get_default ().music_folder = folder;
```
Comment (Mohelm97) [inline src/LocalBackend/LocalLibrary.vala]: I think it's better to save inside the `if` above.

File: src/Dialogs/PreferencesWindow.vala [hunk 2]
```diff
@@ -49,7 +49,8 @@ public class Noise.PreferencesWindow : Gtk.Dialog {
         library_filechooser.hexpand = true;
         library_filechooser.set_current_folder (Settings.Main.get_default ().music_folder);
         library_filechooser.file_set.connect (() => {
-            App.main_window.setMusicFolder (library_filechooser.get_current_folder ());
+            string? filename = library_filechooser.get_uri ().replace ("file://", "");
```
Comment (Mohelm97) [inline src/Dialogs/PreferencesWindow.vala]: Nice catch :D

File: src/LocalBackend/LocalLibrary.vala (hunk 1 above)
Comment (ZavierHenry) [reply src/LocalBackend/LocalLibrary.vala]: Ok I made the changes in the next commit. It seems the Travis CI build failed though and I'm not sure why

Comment (Mohelm97) [reply src/LocalBackend/LocalLibrary.vala]: Travis fails due to some connection problem with ppa.launchpad.net, I will close the PR and reopen it to trigger it to build again.

**Pull request merged**